src/store.c

```diff
@@ -437,6 +439,23 @@ static int _must_use_ _nonnull_ cs_content_add(struct clip_store *cs,
         }
 
         return 0;
+    } else if (dupe && deduplicate) {
+        _drop_(cs_unref) struct ref_guard guard = cs_ref(cs);
+        if (guard.status < 0) {
+            return guard.status;
+        }
+
+        for (int i = 0; i < (int)cs->local_nr_snips; ++i) {
+            if (cs->snips[i].hash == hash) {
+                // move cs->snips[i] to the end of the array
```

Comment:
Good inline comments (as opposed to function/datastructure documentation) generally document why, not how, so not immediately seeing a reason to have this one: the code seems fairly self-explanatory.